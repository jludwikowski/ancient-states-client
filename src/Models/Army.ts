import {Entity} from './Entity';
import Leader from './Leader';
import UnitReference from "./UnitReference";

export default interface Army extends Entity {
    commander: Leader | null,
    units: UnitReference[]
    owner: number,
    position: string,
    status: string,
    orderDate?: string | null,
    targetId?: number | null,
}
