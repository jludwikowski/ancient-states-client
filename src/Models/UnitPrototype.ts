import {Entity} from './Entity';
import {UnitType} from './UnitType';

export default interface UnitPrototype extends Entity {
    baseTrainTime: number,
    baseDamage: number,
    activeTurns: number[],
    baseHealth: number,
    baseCost: number,
    type: UnitType,
    imageUrl: string
}
