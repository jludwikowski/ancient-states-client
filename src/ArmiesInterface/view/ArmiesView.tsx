import React from 'react';
import Army from '../../Models/Army';
import UnitPrototype from '../../Models/UnitPrototype';
import {ArmyCard} from '../components/ArmyCard';
import {ArmyCreationForm} from "../components/ArmyCreationForm";

interface Props {
    armies: Army[],
    baseUnits: UnitPrototype[]
};

interface State {
    creatingArmy: boolean;
}

export class ArmiesView extends React.Component<Props, State> {

    constructor(props:Props) {
        super(props);
        this.state = { creatingArmy: false };
    }

    public onDisband = (id:number | undefined) => {
        console.log('On onDisband');
    };

    public onCreate = () => {
        this.setState( {creatingArmy: true })
    };

    public render () {
        return <>
            <div>
                <ul className="list-group">
                    {Object.values(this.props.armies).map((army) => (
                        <React.Fragment key={army.id}>
                            <ArmyCard army={army} onDisband={this.onDisband} baseUnits={this.props.baseUnits}/>
                        </React.Fragment>
                    ))}
                </ul>
                <div className="card">
                    <button className='btn btn-outline-secondary' onClick={this.onCreate}>
                        Create new Army
                    </button>
                </div>
            </div>
            {this.state.creatingArmy ? <ArmyCreationForm army={this.props.armies[0]} unitPrototypes={this.props.baseUnits} />
                : null }
        </>
    }
}
