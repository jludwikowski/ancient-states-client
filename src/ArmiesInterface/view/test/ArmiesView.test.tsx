import { shallow } from 'enzyme';
import React from 'react';
import ReactDOM from 'react-dom';
import Army from "../../../Models/Army";
import UnitPrototype from "../../../Models/UnitPrototype";
import { ArmiesView } from '../ArmiesView';

describe('ArmiesView', () => {

    const armies:Army[] = [{id: 7,
            name: 'New Army',
            description: '',
            status: 'camp',
            orderDate: null,
            targetId: null,
            position: '0,0',
            owner: 1,
            commander: null,
            units: [
                {
                    number: 9802,
                    level: 1,
                    baseUnit: 1,
                }
            ]}];
    const baseUnits:UnitPrototype[] = [{
        id: 1,
        name: "Persian Spearman",
        description: "Poorly armed and armor eastern warrior",
        baseTrainTime: 50,
        baseDamage: 8,
        baseHealth: 30,
        baseCost: 40,
        activeTurns: [6, 7, 8, 9, 10, 11, 12],
        type: "Melee",
        imageUrl: "images/icons/icons8-military-base-100.png"
    }

    ];

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<ArmiesView armies={armies} baseUnits={baseUnits} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('Should match snapshot', () => {
        const component = shallow(<ArmiesView armies={armies} baseUnits={baseUnits} />);
        expect(component).toMatchSnapshot();
    });
});
