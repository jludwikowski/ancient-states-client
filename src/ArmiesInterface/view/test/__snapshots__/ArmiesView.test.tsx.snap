// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArmiesView Should match snapshot 1`] = `
<Fragment>
  <div>
    <ul
      className="list-group"
    >
      <ArmyCard
        army={
          Object {
            "commander": null,
            "description": "",
            "id": 7,
            "name": "New Army",
            "orderDate": null,
            "owner": 1,
            "position": "0,0",
            "status": "camp",
            "targetId": null,
            "units": Array [
              Object {
                "baseUnit": 1,
                "level": 1,
                "number": 9802,
              },
            ],
          }
        }
        baseUnits={
          Array [
            Object {
              "activeTurns": Array [
                6,
                7,
                8,
                9,
                10,
                11,
                12,
              ],
              "baseCost": 40,
              "baseDamage": 8,
              "baseHealth": 30,
              "baseTrainTime": 50,
              "description": "Poorly armed and armor eastern warrior",
              "id": 1,
              "imageUrl": "images/icons/icons8-military-base-100.png",
              "name": "Persian Spearman",
              "type": "Melee",
            },
          ]
        }
        onDisband={[Function]}
      />
    </ul>
    <div
      className="card"
    >
      <button
        className="btn btn-outline-secondary"
        onClick={[Function]}
      >
        Create new Army
      </button>
    </div>
  </div>
</Fragment>
`;
